docs/book/cloud_provider_interface.md

```diff
@@ -0,0 +1,129 @@
+The use of allowedTopolgies should become less common now that WaitForFirstConsumer is available. However, it is still possible to use allowedTopologies in the StorageClass if there is a desire to control the placement of volumes to certains zones and regions.
+
+Click here to learn more about [Allowed Topologies](https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies).
+
+## How do I get it?
+
+There are 3 components that make up the CPI. These are the cluster roles (RBAC), role bindings and the service, service account and daemonset for running the CPI.
+
+* [Cloud Controlle Manager Roles](https://raw.githubusercontent.com/kubernetes/cloud-provider-vsphere/master/manifests/controller-manager/cloud-controller-manager-roles.yaml)
+* [Cloud Controller Manage Role Bindings](https://raw.githubusercontent.com/kubernetes/cloud-provider-vsphere/master/manifests/controller-manager/cloud-controller-manager-role-bindings.yaml)
+* [Cloud Controller Manager DaemonSet][https://github.com/kubernetes/cloud-provider-vsphere/raw/master/manifests/controller-manager/vsphere-cloud-controller-manager-ds.yaml]
+
+## Which versions of Kubernetes/vSphere support it?
+
+CPI is supported for vSphere versions 6.7U3 or later. CPI also requires a Kubernetes version >= v1.11.
+
+## How do I install it?
+
+The following are a set of steps that must be followed in order to install the vSphere CPI on your Kubernetes cluster.
+
+1) Ensure you have the complete set of docker images required to do the install. For those who use a local repository (because their Kubernetes cluster nodes do not have access to the internet), these images will need to be pulled externally, and pushed to the local repository.
+2) Deploy and configure the Kubernetes master and nodes virtual machines.
+3) Configure the virtual machines, setting adequate CPU and memory resources depending on the workloads planned for the cluster. Use the VMware Paravirtual SCSI controller for the primary disk on the node VMs. Set the disk.EnableUUID parameter for each node VM. This step is necessary so that the VMDK always presents a consistent UUID to the VM, thus allowing the disk to be mounted properly.
+4) Install a Guest OS on all VMs. VMware recommends Ubuntu 18.04.1 LTS (Bionic Beaver) - 64-bit PC (AMD64) Server.
+5)Disable swap on all K8s worker nodes. Use the ‘swapoff -a’ command and then modify the /etc/fstab file and remove any swap entries.
+6) Ensure Docker is installed and running on all nodes. VMware recommends Docker version 18.06 CE.
+7) Install kubeadm version 1.14.2-00 on all master and worker nodes.
+
+Steps 8-10 are specifically for the master(s).
+8) Create an /etc/kubernetes/kubeadminit.yaml cluster configuration file on the master, and run kubeadm init to set up the master node. The manifest states the cloud-provider is external, though we have not yet set it up. If the command is successful, the output will the kubeadm join command to join the worker nodes to the cluster.
```

Comment:
markdown verify doesn't like that this is 8) and think it should be 1) because of the line break.

Reply:
@cormachogan I might convert these to regular bullet points (not numbered) cause it's failing mark down checks.

Reply:
Go for it @andrewsykim

Reply:
@cormachogan, on second thought I think we should revisit this list, there are some thing there that are too prescriptive IMO. For now I'm going to have it point to https://cloud-provider-vsphere.sigs.k8s.io/tutorials/deploying_cloud_provider_vsphere_with_rbac.html and revisit this section later.